providers/implementations/encode_decode/decode_der2key.c

```diff
@@ -107,9 +108,13 @@ static void *der2key_decode_p8(const unsigned char **input_der,
     const X509_ALGOR *alg = NULL;
     void *key = NULL;
 
-    if ((p8inf = d2i_PKCS8_PRIV_KEY_INFO(NULL, input_der, input_der_len)) != NULL
-        && PKCS8_pkey_get0(NULL, NULL, NULL, &alg, p8inf)
-        && OBJ_obj2nid(alg->algorithm) == ctx->desc->evp_type)
+    if ((p8inf = d2i_PKCS8_PRIV_KEY_INFO(NULL, input_der, input_der_len)) !=
+            NULL &&
```

Comment:
The project's code style guide puts the `&&` at the start of the following line.